lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -642,6 +677,10 @@ public boolean unlockAndFreeze(HeldLocksToken token) {
             throw new IllegalArgumentException(MessageFormatter.format(UNLOCK_AND_FREEZE, heldLocks.realToken)
                     .getMessage());
         }
+        if (threadInfoConfiguration.recordThreadInfo()) {
+            // unlockAndFreeze will succeed
```

Comment:
Can you always guarantee this? What happens if updateThreadInfoForUnlock starts failing?